specs/WebNotification.md

```diff
@@ -0,0 +1,747 @@
+  /// This corresponds to the
+  /// [action](https://developer.mozilla.org/docs/Web/API/Notification/actions)
+  /// member of a notification action object.
+  [propget] HRESULT Action([out, retval] LPWSTR* value);
+
+  /// A string containing action text to be shown to the user.
+  /// This corresponds to the
+  /// [title](https://developer.mozilla.org/docs/Web/API/Notification/actions)
+  /// member of a notification action object.
+  [propget] HRESULT Title([out, retval] LPWSTR* value);
+
+  /// A string containing the URI of an icon to display with the action.
+  /// This corresponds to the
+  /// [icon](https://developer.mozilla.org/docs/Web/API/Notification/actions)
+  /// member of a notification action object.
+  [propget] HRESULT IconUri([out, retval] LPWSTR* value);
+}
+
+/// A collection of notification actions.
+[uuid(89D8907E-18C7-458B-A970-F91F645E4C43), object, pointer_default(unique)]
+interface ICoreWebView2NotificationActionCollectionView : IUnknown {
+  /// The number of notification actions contained in the
+  /// ICoreWebView2NotificationActionCollectionView.
+  [propget] HRESULT Count([out, retval] UINT* value);
+
+  /// Gets the notification action at the given index.
+  HRESULT GetValueAtIndex([in] UINT index,
+                          [out, retval] ICoreWebView2NotificationAction** value);
+}
+
+/// A collection of unsigned long integers.
+[uuid(974A91F8-A309-4376-BC70-7537D686533B), object, pointer_default(unique)]
+interface ICoreWebView2UnsignedLongCollection : IUnknown {
+  /// The number of unsigned long integers contained in the
+  /// ICoreWebView2UnsignedLongCollection.
+  [propget] HRESULT Count([out, retval] UINT* value);
+
+  /// Gets the unsigned long integer at the given index.
+  HRESULT GetValueAtIndex([in] UINT index, [out, retval] UINT64* value);
+}
+
+/// The caller implements this interface to receive the result of reporting a notification clicked.
+[uuid(1013C0D5-5F0C-4BF8-BA52-9D76AFA20E83), object, pointer_default(unique)]
+interface ICoreWebView2NotificationReportClickedCompletedHandler : IUnknown {
+    HRESULT Invoke([in] HRESULT errorCode);
+}
+
+/// The caller implements this interface to receive the result of reporting a notification closed.
+[uuid(D8B63F74-1D78-4B4C-ACA9-7CB63FDFD74C), object, pointer_default(unique)]
+interface ICoreWebView2NotificationReportClosedCompletedHandler : IUnknown {
+    HRESULT Invoke([in] HRESULT errorCode);
+}
+
+/// This is the ICoreWebView2Notification that represents a [HTML Notification
+/// object](https://developer.mozilla.org/docs/Web/API/Notification).
+[uuid(E3F43572-2930-42EB-BD90-CAC16DE6D942), object, pointer_default(unique)]
+interface ICoreWebView2Notification : IUnknown {
+  /// Add an event handler for the `CloseRequested` event.
+  /// This event is raised when the notification is closed by the web code, such as 
+  /// through `notification.close()`.
+  HRESULT add_CloseRequested(
```

Comment:
See if we can include this in our new third sample or if it belongs in the other samples.

Good to add that note to the docs that you don't have to call ReportClosed since this is coming from the script.